internal/controller/hubble/hubble.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2025 Illumio, Inc. All Rights Reserved.
+
+package hubble
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+
+	exp_credentials "github.com/illumio/cloud-operator/internal/tls"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/credentials"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	hubbleRelayExpectedServerName        = "ui.hubble-relay.cilium.io"
+	ciliumHubbleRelayServiceName  string = "hubble-relay"
+)
+
+var (
+	ErrCertDataMissingInSecret = errors.New("required certificate data (ca.crt, tls.crt, or tls.key) not found in secret")
+	ErrHubbleNotFound          = errors.New("hubble Relay service not found; disabling Cilium flow collection")
+	ErrNoPortsAvailable        = errors.New("hubble Relay service has no ports; disabling Cilium flow collection")
+)
+
+// getHubbleMTLSCertificatesFromSecret fetches ca.crt, tls.crt (client cert), and tls.key (client key)
+// from the specified Kubernetes secret.
+func getHubbleMTLSCertificatesFromSecret(
+	ctx context.Context,
+	clientset kubernetes.Interface,
+	logger *zap.Logger,
+	secretName string,
+	secretNamespace string,
+) (caData, clientCertData, clientKeyData []byte, err error) {
+
+	logger.Debug("Fetching mTLS certificates from Kubernetes secret",
+		zap.String("name", secretName),
+		zap.String("namespace", secretNamespace))
+
+	secret, err := clientset.CoreV1().Secrets(secretNamespace).Get(ctx, secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to get secret '%s' in namespace '%s': %w", secretName, secretNamespace, err)
+	}
+
+	caData, ok := secret.Data["ca.crt"]
+	if !ok || len(caData) == 0 {
+		return nil, nil, nil, fmt.Errorf("%w: 'ca.crt' key", ErrCertDataMissingInSecret)
+	}
+	logger.Debug("Successfully retrieved ca.crt from secret.")
+
+	clientCertData, ok = secret.Data["tls.crt"]
+	if !ok || len(clientCertData) == 0 {
+		return nil, nil, nil, fmt.Errorf("%w: 'tls.crt' key (client certificate)", ErrCertDataMissingInSecret)
+	}
+	logger.Debug("Successfully retrieved tls.crt (client certificate) from secret.")
+
+	clientKeyData, ok = secret.Data["tls.key"]
+	if !ok || len(clientKeyData) == 0 {
+		return nil, nil, nil, fmt.Errorf("%w: 'tls.key' key (client key)", ErrCertDataMissingInSecret)
+	}
+	logger.Debug("Successfully retrieved tls.key (client key) from secret.")
+
+	return caData, clientCertData, clientKeyData, nil
+}
+
+// loadMTLSCredentialsFromData loads mTLS credentials from byte slice.
+func loadMTLSCredentialsFromData(logger *zap.Logger, aksManaged bool, caCertData, clientCertData, clientKeyData []byte) (credentials.TransportCredentials, error) {
```

Comment:
Find a better name for the boolean flag.
This is not specific to AKS. More generally, it has to do with the Golang version.
```suggestion
func loadMTLSCredentialsFromData(logger *zap.Logger, disableALPN bool, caCertData, clientCertData, clientKeyData []byte) (credentials.TransportCredentials, error) {
```